samtranslator/model/api/api_generator.py

```diff
@@ -287,7 +287,34 @@ def _add_auth(self):
             self._set_default_authorizer(swagger_editor, authorizers, auth_properties.DefaultAuthorizer)
 
         # Assign the Swagger back to template
-        self.definition_body = swagger_editor.swagger
+
+        self.definition_body = self._openapi_auth_postprocess(swagger_editor.swagger)
+
+    def _openapi_auth_postprocess(self, definition_body):
+        """
+        Convert auth components to openapi 3 in definition body if OpenApiVersion flag is specified.
+
+        To maintain backward compatibility with existing swagger documents that contain both swagger
+        and openapi defined in the definition body, we return if there is "swagger" defined in the
+        definition body.
+
+        If the definition body contains "openapi" and not "swagger", we enforce that the OpenApiVersion
+        flag is specified for cases with auth.
+        """
+        if definition_body.get('swagger') is not None:
+            return definition_body
+
+        if definition_body.get('openapi') is not None:
```

Comment:
Some thoughts about this section:

I know we discussed that some people might be adding in `Swagger: '2.0'` as well as `openapi: 3.0.1`  to their templates right now to get around SAM's swagger check. If it turns out that enough people are using this workaround, we want to ship this section of the code.

However, I don't agree with that workaround (and I am also not sure if it actually works). I'll try to verify if it works, and, if it does and it appears as if customers are using that workaround, we can ship this code. If not, I'd rather make the validation in this section tighter so that SAM is a little more explicit about what is supported and what isn't.

Reply:
@keetonian @brettstack If we can ship with this validation in place and later remove it if unnecessary, I would much rather do that. Not sure what you mean by "little more explicit about what is supported and what isn't." @keetonian

Reply:
That's fair.